var/spack/repos/builtin/packages/mfem/package.py

```diff
@@ -1234,8 +1251,9 @@ def cache_test_sources(self):
         install test subdirectory for use during `spack test run`."""
         # Clean the 'examples' directory -- at least one example is always built
         # and we do not want to cache executables.
-        make("examples/clean", parallel=False)
-        cache_extra_test_sources(self, [self.examples_src_dir, self.examples_data_dir])
+        if "+examples" in self.spec:
+            make("examples/clean", parallel=False)
+            cache_extra_test_sources(self, [self.examples_src_dir, self.examples_data_dir])
```

Comment:
The `if` condition is wrong here -- the code here has to be executed regardless of the setting of the variant `examples`. Please revert.